skipruntime-ts/server/src/server.ts

```diff
@@ -5,11 +5,43 @@ import {
 } from "skip-wasm";
 import { createRESTServer } from "./rest.js";
 
+/**
+ * Initialize and start a service.
+ *
+ * Calling `runService` will initialize a service based on the `service` specification and start listening for requests on the specified `port`.
+ * This is a REST service that provides access to the internal reactive service.
+ * The service exposes some resources specified by a {@link SkipService}.
+ * Each resource has a _name_ and identifies a collection that associates _keys_ to _values_.
+ * The service responds to the following HTTP requests:
+ * - Read: `GET /v1/:resource/:key?param1=value1&...&paramN=valueN`
+ *
+ *   Instantiate the named `resource` with parameters `{param1=value1,...,paramN=valueN}` and respond with the values associated with the given `key`. If the named resource has previously been instantiated with the same parameters, then it is reused.
+ *
+ * - Subscribe: `GET /v1/:resource?param1=value1&...&paramN=valueN`
+ *
+ *   Instantiate the named `resource` with parameters `{param1=value1,...,paramN=valueN}`, if necessary, and provide an HTTP server-sent event stream of updates to the resource.
+ *   Each event will be a serialization of a `CollectionUpdate` of the form:
+ * ```
+ *     event: (init | update)\n
+ *     id: <watermark>\n
+ *     data: <values>\n\n
+ * ```
+ *
+ * - Write: `PATCH /v1/:collection`
+ *
+ *   Update the named `collection` with the key-values entries in the request body.
+ *   The `collection` must be the name of one of the service's collections, that is, one of the keys of the `Inputs` type parameter.
```

Comment:
The implementation does not AFAIU check that the collection is one of the service Inputs. But I don't know how user code can come to know the actual name of other collections, such as outputs of resources, in order to issue update requests to them.